truss/templates/shared/logging.py

```diff
@@ -22,11 +22,37 @@ def filter(self, record: logging.LogRecord) -> bool:
         )
 
 
+class StreamToLogger:
+    """
+    StreamToLogger redirects stdout and stderr to logger
+    """
+
+    def __init__(self, logger, log_level, stream):
+        self.logger = logger
+        self.log_level = log_level
+        self.stream = stream
+
+    def __getattr__(self, name):
+        # we need to pass `isatty` from the stream for uvicorn
+        # this is a more general, less hacky fix
+        return getattr(self.stream, name)
+
+    def write(self, buf):
+        for line in buf.rstrip().splitlines():
+            self.logger.log(self.log_level, line.rstrip())
+
+    def flush(self):
```

Comment:
does this not need to be implemented? can you add a comment why?

Reply:
@varunshenoy correct me if I'm wrong here, but I believe flush is a no-op because this is no longer writing to stdout. "write" gets called on each write to stdout, and we immediately write to the logger (without any buffering), which will then deal with any flushing that needs to happen.

Reply:
Correct.

Reply:
Please add that explanation as a comment. it just looks odd to me as a pass without an explanation